server/platform/services/sharedchannel/sync_send_remote.go

```diff
@@ -704,18 +765,26 @@ func (scs *Service) sendSyncMsgToRemote(msg *model.SyncMsg, rc *model.RemoteClus
 	err = rcs.SendMsg(ctx, rcMsg, rc, func(rcMsg model.RemoteClusterMsg, rc *model.RemoteCluster, rcResp *remotecluster.Response, errResp error) {
 		defer wg.Done()
 
-		var syncResp model.SyncResponse
-		if err2 := json.Unmarshal(rcResp.Payload, &syncResp); err2 != nil {
-			scs.server.Log().Log(mlog.LvlSharedChannelServiceError, "Invalid sync msg response from remote cluster",
-				mlog.String("remote", rc.Name),
-				mlog.String("channel_id", msg.ChannelId),
-				mlog.Err(err2),
-			)
+		// Check for ErrChannelNotShared in the response error
+		if errResp != nil && strings.Contains(errResp.Error(), ErrChannelNotShared.Error()) {
```

Comment:
nit: if we return only the `ErrChannelNotShared` as per previous comments, can we use a more robust mechanism like `errors.Is` / `errors.As`?

Reply:
Looking at the ReceiveIncomingMsg handling of the Response structure, seems like error gets converted to a string during the HTTP round trip so not sure we can use smth. like errors.Is. Am I missing smth ?

Reply:
Nope, you're right, I thought we could get the error type easily.